### PR TITLE
Dont throw exception if table metadata update fails

### PR DESCRIPTION
**Pull request by hsaraogi**
**Goals (and why)**: See PDS-61374.

---
This change is [Reviewable](https://reviewable.io/reviews/palantir/atlasdb/2810)

### CODE REVIEW

File: atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueServiceImpl.java [hunk 1]
```diff
@@ -2008,10 +2008,21 @@ private void putMetadataAndMaybeAlterTables(
                 return null;
             });
         } catch (Exception e) {
-            throw Throwables.unwrapAndThrowAtlasDbDependencyException(e);
+            log.error("Failed to apply the table metadata changes for the following Cfdefs {}",
+                    SafeArg.of("cfdef", getAllCfdefsString(updatedCfs)), e);
+
+//            throw Throwables.unwrapAndThrowAtlasDbDependencyException(e);
```
Comment (kyegupov) [inline atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueServiceImpl.java]: ideally should be a config setting "ignoreInternalSchemaUpdateFailures"

File: atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueServiceImpl.java [hunk 2]
```diff
@@ -2008,10 +2008,21 @@ private void putMetadataAndMaybeAlterTables(
...
         }
     }
 
+    private String getAllCfdefsString(Collection<CfDef> updatedCfs) {
```
Comment (kyegupov) [inline atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueServiceImpl.java]: nit: https://docs.oracle.com/javase/8/docs/api/java/util/stream/Collectors.html#joining-java.lang.CharSequence-java.lang.CharSequence-java.lang.CharSequence-

File: atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueServiceImpl.java [hunk 3]
```diff
@@ -2008,10 +2008,21 @@ private void putMetadataAndMaybeAlterTables(
...
+        StringBuilder sb = new StringBuilder("CfDefs(");
```
Comment (tboam) [inline atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueServiceImpl.java]: you're going to get "... Cfdefs CfDefs(...)" which may a look a bit odd - can updated after the major issue is resolved though.

**Comment by gsheasby** [comment]
Should we rebase this to develop? If so, we probably want to retry these metadata changes at some point (e.g. when all C* nodes are up).

**Comment by hsaraogi** [comment]
One concern is is it safe to apply the changes to a subset of nodes and ignore the error. Note that we have already attempted the metadata changes but did not get a schema version agreement from ALL nodes. So, some nodes have the changes.

What happens if other nodes not aware of the new compression block size or gc_grace_seconds or something else in the future start up? Will they get the info from other nodes eventually? are we safe for the short period they are unaware of the changed settings?

**Comment by hsaraogi** [comment]
Also, we should add more information to the message on what were the metadata changes being applied. It was unclear in PDS-61374 as to what the metadata updates were.

**Comment by hsaraogi** [comment]
Closing, as this was a proposed band-aid and we wont be merging it to develop.

**Pull request closed without merging**